rueidisprob/slidingbloomfilter_retry.go

```diff
@@ -0,0 +1,37 @@
+//go:build !go1.22
+// +build !go1.22
+
+package rueidisprob
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+const (
+	defaultMaxRetryDelay = time.Second
+)
+
+var rngPool = sync.Pool{
+	New: func() any {
+		return rand.New(rand.NewSource(time.Now().UnixNano()))
+	},
+}
+
+func fastRand(n int) (r int) {
+	s := rngPool.Get().(*rand.Rand)
+	r = s.Intn(n)
+	rngPool.Put(s)
+	return
+}
+
+// Default backoff implementation
+func defaultBackoff(attempts int, maxRetries int) (time.Duration, bool) {
+	if attempts >= maxRetries {
+		return 0, false
+	}
+	base := 1 << min(maxRetries, attempts)
+	jitter := fastRand(base)
```

Comment:
I guess you can just use the `util.FastRand` from the `internal/util` package and no need to have a separated slidingbloomfilter_retry.1.22.go